Sources/OpenGoogleSignInSDK/OpenGoogleSignInSDK.swift

```diff
@@ -148,7 +154,43 @@ public final class OpenGoogleSignIn: NSObject {
             return
         }
         
-        let task = session.dataTask(with: tokenRequest) { data, response, error in
+        makeRequest(tokenRequest) { result in
+            switch result {
+            case let .success(data):
+                do {
+                    var user = try self.decodeUser(from: data)
+                    
+                    guard let profileRequest = self.makeProfileRequest(user: user) else {
+                        completion(.success(user))
+                        return
+                    }
+                    
+                    self.makeRequest(profileRequest) { result in
+                        switch result {
+                        case let .success(data):
+                            let decoder = JSONDecoder()
+                            decoder.keyDecodingStrategy = .convertFromSnakeCase
```

Comment:
I think this might be another discussion, if we should make new decoder for each request/decoding or use a shared one. Currently this decoder and the [one used in `decodeUser(...)`](https://github.com/AckeeCZ/OpenGoogleSignInSDK/blob/0a85973547f1ae9973aee8bec1308a06d9b7b565/Sources/OpenGoogleSignInSDK/OpenGoogleSignInSDK.swift#L136) are configured in exactly the same way, but that shouldn't automatically mean, that they should be merged into single shared instance.

Can we discuss that?

Reply:
I merged them. For me the main argument is that we are working with one API and therefore I hope there is some consistency between requests